app/server.py

```diff
@@ -0,0 +1,90 @@
+from sanic import Sanic
+from sanic.request import Request
+from sanic.response import redirect, json
+from sanic.log import logger
+
+from utils import get_config, AuthConfig
+from auth import Auth
+
+app = Sanic(name="backend")
+
+app.ctx.cfg = get_config(logger)
+
+ac = AuthConfig(app.ctx.cfg)
+
+def respond_update_cookie(request: Request, response, cookie: str):
+    """
+    Do the right things when writing a user cookie
+    along with a response.
+    """
+    response.cookies["user"] = cookie
+    response.cookies['user']['httponly'] = True
+    response.cookies['user']['secure'] = True
+    
+    # Note - split as it needs to be "localhost" not "localhost:<port>"
+    # TODO - use a library to make url, not nasty splits
+    response.cookies['user']['domain'] = f'{request.host.split(":")[0]}'
+
+    return response
+
+
+@app.route('/')
+async def home(request: Request):
+    """
+    Super simple endpoint, returns some json
+    """
+    auth = Auth(ac, request, logger)
+    return json({"you're logged in as an admin": auth.has_admin()}, status=200)
```

Comment:
Yeah, as I think you've been mentioning it'd be good to be able to use a decorator like `@authorize(roles=["admin"])` to ensure only permitted individuals can access protected end-points.

Reply:
@thomasmastersmethods if you find yourself with some spare time this afternoon, do you want to try giving this a go?

Reply:
Yes, if I get a window certainly take a look at it